lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -77,45 +106,155 @@ def __init__(
+            authorized_id,
+            user,
+            access,
+            is_admin,
+        )
+
+        access_term = None
+        user_term = None
+
+        # If a user is specified, assume we'll be selecting for that user
         if user:
-            term = {"authorization.owner": user}
-        elif json.get("access") == Dataset.PRIVATE_ACCESS:
-            raise UnsupportedAccessMode(user, Dataset.PRIVATE_ACCESS)
+            user_term = {"term": {"authorization.owner": user}}
+
+        # IF we're looking at a user we're not authorized to see (either the
+        # call is unauthenticated, or authenticated as a non-ADMIN user trying
+        # to reference a different user), then we're only allowed to see PUBLIC
+        # data; private data requests aren't allowed, and we expect these cases
+        # to fail with appropriate permission errors (401 or 403) before we
+        # reach this code.
+        #
+        # ELSE IF we're looking for a specific access category, add the query
+        # term.
+        #
+        # ELSE IF this is a {} query from a non-ADMIN user, add a "disjunction"
+        # (OR) covering all the user's own data plus all public data.
+        #
+        # ELSE this must be an admin user looking for data owned by a specific
+        # user (there's no access parameter, and possibly no user parameter),
+        # and we'll pass through the query either with the specified user
+        # constraint or with no constraint at all.
+        if not authorized_user or (user and user != authorized_id and not is_admin):
+            if access == Dataset.PRIVATE_ACCESS:
+                requestor = (
+                    f"Requestor {authorized_user.username!r}"
+                    if authorized_user
+                    else "Unauthorized requestor"
+                )
+                raise AssemblyError(
+                    f"Internal error: {requestor} can't query private data"
+                )
+            access_term = {"term": {"authorization.access": Dataset.PUBLIC_ACCESS}}
+            self.logger.debug("QUERY: not self public: {}", filter)
+        elif access:
+            access_term = {"term": {"authorization.access": access}}
+            if not user and access == Dataset.PRIVATE_ACCESS and not is_admin:
+                user_term = {"term": {"authorization.owner": authorized_id}}
+            self.logger.debug("QUERY: access: {}", filter)
+        elif authorized_user and not user and not is_admin:
+            filter.append(
+                {
+                    "dis_max": {
+                        "queries": [
+                            {"term": {"authorization.owner": authorized_id}},
+                            {"term": {"authorization.access": Dataset.PUBLIC_ACCESS}},
+                        ]
+                    }
+                }
+            )
+            self.logger.debug("QUERY: {{}} default: {}", filter)
+            return {"bool": {"filter": filter}}
```

Comment:
Can't we just use the return at line 257?  By the time we get to 245, `access_term` is still `None`, and `user_term` is must be `None` (`not user` boolean in this `elif`).